### ISSUE TITLE
Implement an indeterminate loading UI

### ISSUE THREAD

**Issue by jaril** [RecordReplay/devtools#3523]
Our loading screen should be flexible enough to handle different states without needing to reload based on different if statements.

---

Zypsy passed along a loading animation. Can we please put this on our loading screen?
[loader.html.zip](https://github.com/RecordReplay/devtools/files/7335553/loader.html.zip)

**Comment by jonbell-lot23** [comment]
@jaril and @jasonLaster, based on our discussion today it sounds like this isn't really a Rive issue. Instead it's a loading screen issue.

Today we decided that that refactoring should be done, so it's just a matter of how important it is from a design POV. My perspective is that we don't need to do it this week or next ... but we should do it this quarter.

I propose we close out this particular ticket and make a new one that talks about refactoring the loading screen.

**Comment by jasonLaster** [comment]
Sounds good. Updated the title

**Comment by jonbell-lot23** [comment]
CC @jasonLaster @jaril @gideonred 

Are we ready to tackle this? It's getting in the way of some important first UX/impression work, so I'd like us to bite the bullet and refactor our loading screen soon.

**Comment by jasonLaster** [comment]
@jonbell-lot23 the big blocker was switching to Next.js, so it is now technically unblocked, but we should also check-in on the onboarding funnel this week. My hunch is that there will also be quite a bit of Next.js fallout as well.

**Comment by jonbell-lot23** [comment]
That's fair @jasonLaster . Here's my proposal:

* Onboarding + bugs are highest priority, of course
* This is next highest priority

That means we move forward on this without getting in the way of onboarding. How's that sound?

**Comment by jonbell-lot23** [comment]
We know from design psychology that a loading screen that looks more professional will a) feel faster b) make people less judgemental when things go wrong. So between this ticket and #4541 this is work we need to address. Ping!

**Comment by jasonLaster** [comment]
Noted. @jaril what do you think it would take to show some kind of lottie spinner

btw @jonbell-lot23 what do you think of simplifying the animation down to the core essentials? https://www.figma.com/file/SSr1ljyzF0lCXTL7BX1Whh/Replay-Design-Doc?node-id=9320%3A98987

**Comment by jonbell-lot23** [comment]
Don't love it -- we've had that design for ages and I prefer the new ones. Partially because I think they look better, partially because I don't want to re-visit all those hours of animation discussion.

But I care less about the specific animation and more that we refactor the loading screen and get *something* shipped live.

**Comment by jonbell-lot23** [comment]
And so this doesn't got lost in the shuffle, note that we also have the privacy loading screen revision queued up and ready to go.

So even if we don't have the animation bit ready to go, we can at least address the concerns we have about the current privacy/loader.

**Comment by jonbell-lot23** [comment]
Here's the privacy+loading ticket I mentioned, which is more important than the animation bit:
https://github.com/RecordReplay/devtools/issues/4541

**Comment by jasonLaster** [comment]
closing in favor of #4541. We can revisit a more generic spinner later
